### DISCUSSION TITLE
Autosave indicator

### DISCUSSION THREAD

**Discussion by bmartin1134** [vazco/uniforms#891] [Q&A] [answered]
An option on the form to add an indicator as to the status of the autosave.
I imagine adding 3 states properties that can be updated by the onSubmit function or a callback within the onSubmit function:
[Checkmark icon] + [String: Saved]
[Spinning icon] + [String: Updating]
[Error icon] + [String: Error]

I was able to achieve anyways but it would be a nice feature.

**Comment by radekmie** [comment, marked as answer]
Hi @bmartin1134 and thanks for the suggestion. While I don't think there's a need for it in the core themes (what loader to show, maybe you'd need text there, whether it's at the top or the bottom, is the form disabled while it's saving, etc.), I agree that it may be a good example in our docs.

Here's a quick example of how it can be used:
```tsx
import React from 'react';
import { useForm } from 'uniforms';

enum FormState {
  Error,
  Idle,
  Submitting,
  Validating,
}

function useFormState() {
  const uniforms = useForm();
  return uniforms.error
    ? FormState.Error
    : uniforms.submitting
    ? FormState.Submitting
    : uniforms.validating
    ? FormState.Validating
    : FormState.Idle;
}

function FormState() {
  const state = useFormState();
  return <div>{`Current form state is: ${state}`}</div>;
}
```

**Reply by bmartin1134** [reply]
This is perfect thanks for pointing me in the right direction.
